File: week7/binary_search.c
#include <stdio.h>

int binary_search(int arr[], int size, int target) {
    int left = 0;
    int right = size - 1;
    while (left <= right) {
        int mid = (left + right)/2;
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1;

}

int main () {

    int numbers[5] = {3, 8, 12, 15, 27};
    int size = 5;
    int target;

    printf("Enter a number to search for: ");
    scanf("%d", &target);

    int index = binary_search(numbers, size, target);

    if (index == -1) {
        printf("Value %d not found.\n", target);
    } else {
        printf("Value %d found at index %d.\n", target, index);
    }


    return 0;
}

File: week7/binary_search.py
def binary_search(lst, target):
    left = 0
    right = len(lst) - 1
    while left <= right:
        mid = (left + right)//2
        if lst[mid] == target:
            return mid
        elif lst[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1      

def main():
    numbers = [8, 12, 25, 34, 44]
    print("Sorted list:", numbers)

    target = int(input("Enter a number to search for: "))
    idx = binary_search(numbers, target)

    if idx == -1:
        print(f"Value {target} not found")
    else:
        print(f"Value {target} found at index {idx}")

if __name__ == "__main__":
    main()